sdk/search/Azure.Search.Documents/src/Models/SynonymMap.cs

```diff
@@ -0,0 +1,78 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SynonymMap"/> class.
+        /// </summary>
+        /// <param name="name">The name of the synonym map.</param>
+        /// <param name="synonym">The synonym to define.</param>
+        /// <param name="format">The format of the synonym map. Currently, only "soln" is supported and is the default value.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/>, <paramref name="synonym"/>, or <paramref name="format"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/>, <paramref name="synonym"/>, or <paramref name="format"/> is null.</exception>
+        public SynonymMap(string name, string synonym, string format = DefaultFormat)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+            Argument.AssertNotNullOrEmpty(format, nameof(format));
+            Argument.AssertNotNullOrEmpty(synonym, nameof(synonym));
+
+            Name = name;
+            Format = format;
+            Synonyms = synonym;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SynonymMap"/> class.
+        /// </summary>
+        /// <param name="name">The name of the synonym map.</param>
+        /// <param name="synonyms">One or more synonyms to define. These values will be separated by line breaks automatically.</param>
+        /// <param name="format">The format of the synonym map. Currently, only "soln" is supported and is the default value.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/>, <paramref name="synonyms"/>, or <paramref name="format"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/>, <paramref name="synonyms"/>, or <paramref name="format"/> is null.</exception>
+        public SynonymMap(string name, IEnumerable<string> synonyms, string format = DefaultFormat)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+            Argument.AssertNotNullOrEmpty(format, nameof(format));
+            Argument.AssertNotNullOrEmpty(synonyms, nameof(synonyms));
+
+            Name = name;
+            Format = format;
+            Synonyms = string.Join(NewLine, synonyms);
+        }
+
+        private SynonymMap()
+        {
+        }
+
+        /// <summary>
+        /// Gets the name of the synonym map.
+        /// </summary>
+        [CodeGenSchemaMember("name")]
+        public string Name { get; internal set; }
+
+        /// <summary>
+        /// Gets the format of the synonym map.
+        /// </summary>
+        [CodeGenSchemaMember("format")]
+        public string Format { get; internal set; }
+
+        /// <summary>
+        /// Gets the synonym rules for this synonym map.
+        /// </summary>
+        [CodeGenSchemaMember("synonyms")]
+        public string Synonyms { get; internal set; }
```

Comment:
We've exposed things like OrderBy as a collection of values, not the joined string.

Reply:
I've talked with @brjohnstmsft and folks about this and it serializes as a string, whether delimited or not. While conceptually I agree with you, the service team does not want this and the string is opaque passed to the server anyway. Strings may not always be delimited anyway.

If you agree with my original suggestion in bug #10725 for the parameter (and, by extension, this property) please chime in on the discussion in our internal channel. I think what we're suggesting is more SDK-ish.

Reply:
Doesn't the parameter come in as IEnumerable now?  I guess that's my confusion.  We're making the change in one place but not the other.  I agree a collection is easier to work with than a joined string though.